docs/developer/developer-manual.html

```diff
@@ -161,6 +161,26 @@ <h2 id="Build_tasks">Build tasks</h2>
 main directory or any subproject runs the <code>allTests</code> task only in the <code>framework</code> project.
 </p>
 
+<p>
+An iterative build-test-debug cycle where you assemble the Checker Framework
+and execute its tests can sometimes lead to a state where Gradle is not able to
+compile and build the system for no obvious reason.
+This is especially common when you kill tests before they fully finish, or kill
+compilation before it is fully complete, leading to a bad state or an
+inconsistent build cache.
+</p>
+
+<p>
+Below are some steps you might try to resolve this issue (in order):
+</p>
+<ul>
+  <li><code>./gradlew --no-build-cache assemble</code>: attempts to rebuild the Checker Framework without the build cache.
+  <li><code>git clean -fdx</code>: removes all files that are not under version control. <b>Add or stash files that you would like to
+      keep before running this command</b>.
+  <li>Kill any Gradle daemons; run <code>./gradlew --status</code> to get a list of pids for Gradle.
```

Comment:
I think it would be best to list this one before `git clean -fdx`.  `git clean -fdx` is a more drastic step that will delete things like some of your IntelliJ IDE state, so better to try killing all the daemons first.  I personally might even try killing the daemons before going for the `--no-build-cache` trick; my feeling is messed-up daemons are more likely than a corrupted build cache (though maybe for Checker Framework things are different).